sonic_y_cable/y_cable_base.py

```diff
@@ -0,0 +1,577 @@
+        """
+        This API specifically returns which type of switching mode the cable is set to auto/manual
+
+        Args:
+             physical_port:
+                 an Integer, the actual physical port connected to Y end of a Y cable which can toggle the MUX
+
+        Returns:
+            an Integer, SWITCHING_MODE_AUTO if auto switch is enabled.
+                        SWITCHING_MODE_MANUAL if manual switch is enabled.
+        """
+
+        raise NotImplementedError
+
+    def get_nic_temperature(self, port):
+        """
+        This API specifically returns nic temperature of the physical port specified
+
+        Args:
+             physical_port:
+                 an Integer, the actual physical port connected to Y end of a Y cable which can toggle the MUX
+
+        Returns:
+            an Integer, the temperature of the NIC MCU
+        """
+
+        raise NotImplementedError
+
+    def get_local_temperature(self, port):
+        """
+        This API specifically returns local ToR temperature of the physical port specified
+
+        Args:
+             physical_port:
+                 an Integer, the actual physical port connected to Y end of a Y cable which can toggle the MUX
+
+        Returns:
+            an Integer, the temperature of the local MCU
+        """
+
+        raise NotImplementedError
+
+    def get_nic_voltage(self, port):
+        """
+        This API specifically returns nic voltage of the physical port specified
+
+        Args:
+             physical_port:
+                 an Integer, the actual physical port connected to Y end of a Y cable which can toggle the MUX
+
+        Returns:
+            a float, the voltage of the NIC MCU
+        """
+
+        raise NotImplementedError
+
+    def get_local_voltage(self, port):
+        """
+        This API specifically returns local ToR voltage of the physical port specified
+
+        Args:
+             physical_port:
+                 an Integer, the actual physical port connected to Y end of a Y cable which can toggle the MUX
+
+        Returns:
+            a float, the voltage of the local MCU
+        """
+
+        raise NotImplementedError
+
+    def enable_prbs_mode(self, port, target, mode_value, lane_map):
+        """
+        This API specifically configures and enables the PRBS mode/type depending upon the mode_value the user provides.
+        The mode_value configures the PRBS Type for generation and BER sensing on a per side basis.
+        Target is an integer for selecting which end of the Y cable we want to run PRBS on.
+        LaneMap specifies the lane configuration to run the PRBS on.
+        Note that this is a diagnostic mode command and must not run during normal traffic/switch operation
+
+        Args:
+            physical_port:
+                an Integer, the actual physical port connected to a Y cable
+            target:
+                an Integer, the target on which to enable the PRBS
```

Comment:
Maybe rather than explicitly stating "An Integer", we simply state "One of the following predefined constants:". This would also allow us to change the underlying type transparently if desired.

Reply:
fixed